build/admission/gen-admission-secret.sh

```diff
@@ -32,24 +32,52 @@ DNS.2 = ${SERVICE}.${NAMESPACE}
 DNS.3 = ${SERVICE}.${NAMESPACE}.svc
 EOF
 
-openssl genrsa -out ${CERTDIR}/ca.key 2048
-openssl req -x509 -new -nodes -key ${CERTDIR}/ca.key -subj "/CN=${SERVICE}.${NAMESPACE}.svc" -out ${CERTDIR}/ca.crt
+  openssl genrsa -out ${CERTDIR}/ca.key 2048
+  openssl req -x509 -new -nodes -key ${CERTDIR}/ca.key -subj "/CN=${SERVICE}.${NAMESPACE}.svc" -out ${CERTDIR}/ca.crt
 
-openssl genrsa -out ${CERTDIR}/server.key 2048
-openssl req -new -key ${CERTDIR}/server.key -subj "/CN=${SERVICE}.${NAMESPACE}.svc" -out ${CERTDIR}/server.csr -config ${CERTDIR}/csr.conf
+  openssl genrsa -out ${CERTDIR}/server.key 2048
+  openssl req -new -key ${CERTDIR}/server.key -subj "/CN=${SERVICE}.${NAMESPACE}.svc" -out ${CERTDIR}/server.csr -config ${CERTDIR}/csr.conf
 
-openssl x509 -req -in  ${CERTDIR}/server.csr -CA  ${CERTDIR}/ca.crt -CAkey  ${CERTDIR}/ca.key \
--CAcreateserial -out  ${CERTDIR}/server.crt \
--extensions v3_req -extfile  ${CERTDIR}/csr.conf
+  openssl x509 -req -in  ${CERTDIR}/server.csr -CA  ${CERTDIR}/ca.crt -CAkey  ${CERTDIR}/ca.key \
+  -CAcreateserial -out  ${CERTDIR}/server.crt \
+  -extensions v3_req -extfile  ${CERTDIR}/csr.conf
+}
 
-if [[ "${ENABLE_CREATE_SECRET}" = true ]]; then
-    kubectl get ns ${NAMESPACE} || kubectl create ns ${NAMESPACE}
+function createObjects() {
+  # `ENABLE_CREATE_SECRET` should always be set to `true` unless it has been already created.
+  if [[ "${ENABLE_CREATE_SECRET}" = true ]]; then
+      kubectl get ns ${NAMESPACE} || kubectl create ns ${NAMESPACE}
 
-    # create the secret with CA cert and server cert/key
-    kubectl create secret generic ${SECRET} \
-        --from-file=tls.key=${CERTDIR}/server.key \
-        --from-file=tls.crt=${CERTDIR}/server.crt \
-        --from-file=ca.crt=${CERTDIR}/ca.crt \
-        --dry-run -o yaml |
-    kubectl -n ${NAMESPACE} apply -f -
-fi
+      # create the secret with CA cert and server cert/key
+      kubectl create secret generic ${SECRET} \
+          --from-file=tls.key=${CERTDIR}/server.key \
+          --from-file=tls.crt=${CERTDIR}/server.crt \
+          --from-file=ca.crt=${CERTDIR}/ca.crt \
+          -n ${NAMESPACE}
+  fi
+
+  # create other objects needed for admission
+  for resource in $(ls *.yaml); do kubectl create -f $resource; done
```

Comment:
i'd prefer split the deploy and cert generate in different scripts, and in the long run, we should centralize the deployment, using helm or keadm.

Reply:
reverted the change here.